fog/overseer/server/tests/inactive_outstanding_key_idle_does_not_have_key.rs

```diff
@@ -0,0 +1,216 @@
+    );
+    let (node1, _state_file_node_1, client_listen_uri1) = utils::make_node(
+        1,
+        peer_indices.iter().cloned(),
+        recovery_db.clone(),
+        &watcher_path,
+        &ledger_db_path,
+        logger.clone(),
+    );
+    let (node2, _state_file_node2, client_listen_uri2) = utils::make_node(
+        2,
+        peer_indices.iter().cloned(),
+        recovery_db.clone(),
+        &watcher_path,
+        &ledger_db_path,
+        logger.clone(),
+    );
+
+    assert!(!node0.is_active());
+    assert!(!node1.is_active());
+    assert!(!node2.is_active());
+
+    // Give RPC etc. time to start
+    std::thread::sleep(Duration::from_secs(1));
+
+    node0.activate().expect("node0 failed to activate");
+
+    assert!(node0.is_active());
+    assert!(!node1.is_active());
+    assert!(!node2.is_active());
+
+    let first_node0_ingress_key = node0.get_ingest_summary().get_ingress_pubkey().clone();
+    let first_node1_ingress_key = node1.get_ingest_summary().get_ingress_pubkey().clone();
+    let first_node2_ingress_key = node2.get_ingest_summary().get_ingress_pubkey().clone();
+
+    // At first, all the nodes should share the same ingress key.
+    assert_eq!(first_node0_ingress_key, first_node1_ingress_key);
+    assert_eq!(first_node0_ingress_key, first_node2_ingress_key);
+    assert_eq!(first_node1_ingress_key, first_node2_ingress_key);
+
+    // Change the ingress keys on node1 and node2 so that they're different than
+    // node0's ingress key, which is the currently active key.
+    node1.set_new_keys().unwrap();
+    node2.set_new_keys().unwrap();
+
+    let first_node0_ingress_key = node0.get_ingest_summary().get_ingress_pubkey().clone();
+    let first_node1_ingress_key = node1.get_ingest_summary().get_ingress_pubkey().clone();
+    let first_node2_ingress_key = node2.get_ingest_summary().get_ingress_pubkey().clone();
+
+    assert_ne!(first_node0_ingress_key, first_node1_ingress_key);
+    assert_ne!(first_node0_ingress_key, first_node2_ingress_key);
+    assert_ne!(first_node1_ingress_key, first_node2_ingress_key);
+
+    // Initialize an OverSeerService object
+    let mut overseer_service = OverseerService::new(
+        vec![client_listen_uri0, client_listen_uri1, client_listen_uri2],
+        recovery_db.clone(),
+        logger.clone(),
+    );
+    overseer_service.start().unwrap();
+
+    // Set up the Rocket instance
+    let overseer_state = OverseerState { overseer_service };
+    // TODO: Consider testing the CLI here instead.
+    let rocket_config: rocket::Config =
+        rocket::Config::build(rocket::config::Environment::Development)
+            // TODO: Make these either passed from CLI or in a Rocket.toml.
+            .address("127.0.0.1")
+            .port(80)
+            .unwrap();
+    let rocket = server::initialize_rocket_server(rocket_config, overseer_state);
+    let client = Client::new(rocket).expect("valid rocket instance");
+    let _req = client.post("/arm").dispatch();
```

Comment:
I'm surprised to see this code in the tests. Why can't it use the stuff inside `fog/overseer/server/src/server.rs`, or just operate on `overseer_service` directly?

Reply:
I'd like to make it a proper end to end test. Why not exercise all of the code paths, rather than a layer beneath it? Without this, we can't be certain that the rocket logic is working correctly.